### PR TITLE
Prepare for backend connector

### PR DESCRIPTION
**Pull request by ggwpez**
Improvements to make everything work with the updated node version and the backend connector.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -234,7 +234,7 @@ pub mod pallet {
 		/// Disputes a channel in case of a dishonest participant.
 		///
 		/// Can only be called with a non-finalized state that is signed by
-		/// all participants.
+		/// all participants when there is currently no on-going dispute.
```
Comment (matthiasgeihs) [inline src/lib.rs:237]:
```suggestion
		/// all participants if there is no on-going dispute.
```

File: Cargo.toml [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 authors = ["PolyCrypt GmbH <info@polycry.pt>"]
+description = "FRAME pallet for Perun State Channels"
 edition = "2018"
-license = "Apache-2.0"
 homepage = "https://polycry.pt/"
 keywords = ["blockchain", "channel", "perun"]
-repository = "https://github.com/perun-network/perun-polkadot-pallet"
-description = "FRAME pallet for Perun State Channels"
+license = "Apache-2.0"
+name = "pallet-perun"
```
Comment (matthiasgeihs) [inline Cargo.toml:8]: Why did you reorder? I think the ordering before appeared more natural, name and version at top.

File: src/types.rs [hunk 3]
```diff
@@ -168,3 +176,14 @@ where
 		sig.verify(&*msg, &self.part)
 	}
 }
+
+impl<ChannelId, PK> Funding<ChannelId, PK>
+where
+	Funding<ChannelId, PK>: Encode,
+{
+	/// Calculates the funding id of a participant in a channel.
+	pub fn id<H: Hasher>(&self) -> H::Out {
```
Comment (matthiasgeihs) [inline src/types.rs:193]: Where and how is the funding type used? I don't see any usages in this commit.

File: src/types.rs [hunk 4]
```diff
@@ -167,14 +167,22 @@ where
...
+	ChannelId: Clone,
 {
 	/// Returns whether `sig` is a valid signature for this Withdrawal
 	/// and was created by the participant that wants to claim the funds.
 	pub fn validate_sig<Sig: Verify<Signer = Pk>>(&self, sig: &Sig) -> bool {
 		let msg = Encode::encode(&self);
 		sig.verify(&*msg, &self.part)
 	}
+
+	pub fn funding(&self) -> Funding<ChannelId, Pk> {
```
Comment (matthiasgeihs) [inline src/types.rs:180]: You are converting a withdrawal into a funding? That sounds odd.

File: src/lib.rs [hunk 5]
```diff
@@ -197,7 +197,7 @@ pub mod pallet {
 		#[pallet::weight(10_000)]
 		/// Deposits funds for a participant into a channel.
 		///
-		/// The `funding_id` is calculated with [Pallet::calc_funding_id].
+		/// The `funding_id` is calculated with [Funding::id].
```
Comment (matthiasgeihs) [inline src/lib.rs:200]: Isn't it `Funding.id`?

File: src/lib.rs [hunk 6]
```diff
@@ -467,8 +461,8 @@ impl<T: Config> Pallet<T> {
 		let mut sum_outcome = BalanceOf::<T>::default();
 		let mut sum_deposit = BalanceOf::<T>::default();
 
-		for (i, part) in parts.iter().enumerate() {
-			let fid = Self::calc_funding_id(channel, part);
+		for (i, part) in parts.iter().cloned().enumerate() {
+			let fid = FundingOf::<T> { channel, part }.id::<HasherOf<T>>();
```
Comment (matthiasgeihs) [inline src/lib.rs:465]: I think I like using the explicit method better than creating a funding object here. I suggest you keep the method and do the respective operation there.

File: tests/common/mock.rs [hunk 7]
```diff
@@ -172,8 +172,16 @@ pub fn new_setup() -> Setup {
 			carl: keys[2].clone(),
 		},
 		fids: FIDs {
-			alice: Perun::calc_funding_id(cid, &params.participants[0]),
-			bob: Perun::calc_funding_id(cid, &params.participants[1]),
+			alice: FundingOf::<Test> {
```
Comment (matthiasgeihs) [inline tests/common/mock.rs:176]: Please keep `calc_funding_id` and revert the change. It looked way nicer before. You can create the funding object within `calc_funding_id`.

File: tests/common/utils.rs [hunk 8]
```diff
@@ -28,31 +28,31 @@ use sp_core::{crypto::*, H256};
 pub fn event_deposited(funding_id: H256, amount: u64) {
 	assert_eq!(
 		last_event(),
-		Event::pallet_perun(pallet_perun::Event::Deposited(funding_id, amount))
+		Event::Perun(pallet_perun::Event::Deposited(funding_id, amount))
```
Comment (matthiasgeihs) [inline tests/common/utils.rs:31]: Can you explain the reason behind this change? Where do you define `Perun`?

**Review by matthiasgeihs — CHANGES_REQUESTED** [review]
When trying to build I get: ```error: failed to select a version for `bitflags`.```

Any ideas?

File: Cargo.toml (hunk 2 above)
Comment (ggwpez) [reply Cargo.toml:8]: The TOML formatter that i used ordered them alphabetically.

File: src/types.rs (hunk 3 above)
Comment (ggwpez) [reply src/types.rs:193]: It is only introduced in that commit, the transition from `calc_funding_id` to `Funding.id()` happens in the next commit.

File: src/types.rs (hunk 4 above)
Comment (ggwpez) [reply src/types.rs:180]: Hm yes the naming is not intuitive. The `Funding` is only used to calculate the ID, i will think of a better name.

File: src/lib.rs (hunk 5 above)
Comment (ggwpez) [reply src/lib.rs:200]: The signature of the function is `Funding::id ` since it is a member function.  
`Funding.id` would be a member function access, not a type.  
The CI checks for all doc references so you dont need to worry about them being invalid :wink: .

File: src/lib.rs (hunk 6 above)
Comment (ggwpez) [reply src/lib.rs:465]: Yes i could also do that, it probably reads easier.  
I will still keep the `Funding` (maybe rename) type since i need it in the connector.

File: tests/common/utils.rs (hunk 8 above)
Comment (ggwpez) [reply tests/common/utils.rs:31]: This change came from the update `3.0.0` -> `4.0.0-dev` from frame.  
The frame configuration changed and the aliases from `mock.rs` can now be used to address the pallet.

File: src/lib.rs (hunk 1 above)
Comment (ggwpez) [reply src/lib.rs]: I just saw that this is actually rebased onto #3, fixed it there instead.

File: Cargo.toml (hunk 2 above)
Comment (matthiasgeihs) [reply Cargo.toml:8]: OK, I don't know. I think in this case ordering by (subjective) relevance would be more suitable. Is that a special kind of formatter you are using or a common thing to do in rust projects?

**Review by matthiasgeihs — APPROVED** [review]
There is one minor open discussion on the ordering of the items in Cargo.toml. Up to you.

**Comment by ggwpez** [comment]
>When trying to build I get: error: failed to select a version for `bitflags`.

I dont know. Maybe you are using a different rust version?

File: Cargo.toml (hunk 2 above)
Comment (ggwpez) [reply Cargo.toml]: Not that i know of. Just a normal TOML formatter. I reverted the changes.

**Comment by matthiasgeihs** [comment]
> > When trying to build I get: error: failed to select a version for `bitflags`.
> 
> I dont know. Maybe you are using a different rust version?

I am using the latest, version `rustc 1.55.0`. Can you upgrade and check again? I would rather not merge before we understand the issue.

**Review by matthiasgeihs — APPROVED** [review]
